app/Models/Collection.php

```diff
@@ -417,4 +417,22 @@ public function recentlyViewed(): bool
 
         return $this->last_viewed_at->gte(now()->subDays(1));
     }
+
+    /**
+     * A collection is only *potentially* full if the total supply matches the last token number that was indexed for the collection.
+     *
+     * Or if the number of NFTs in a collection matches the total supply.
+     */
+    public function isPotentiallyFull(): bool
+    {
+        if ($this->last_indexed_token_number === null || $this->supply === null) {
+            return false;
+        }
+
+        if ((string) $this->supply === (string) $this->last_indexed_token_number) {
```

Comment:
this is a tricky check, because the token numbers often (not always) start from 0, causing a one-off issue here. There are also various NFTs that don't have a logical numbering for their tokens, for example `AlphaDogs` (address: `0x32f8EE2B5707138e1bdd04D3631A04EB104dc141`, supply is 4625 but their token numbers are strings like `77699296229261824`)